package.json

```diff
@@ -21,8 +21,8 @@
     "test:ci": "npm run test:unit",
     "test:coverage": "jest --config=tests/unit/jest.config.js --coverage",
     "test:firefox": " npm run debug && jest --config=tests/browser/jest.config.firefox.js --runInBand",
-    "test:inject": "node --max-old-space-size=3072 tests/inject/run.js",
-    "test:inject:debug": "node --max-old-space-size=3072 tests/inject/run.js --debug",
+    "test:inject": "node --max-old-space-size=3072 node_modules/.bin/karma start ./tests/inject/karma.conf.js",
+    "test:inject:debug": "node --max-old-space-size=3072 node_modules/.bin/karma start ./tests/inject/karma.conf.js --debug",
```

Comment:
Onto the problem with failing tests returning a zero and leaving the CI test run green.

The signals and cause and effect was getting confusing in `./tests/inject/run.js`, with `process`, `karmaServer`, and `echoServer` all being eventEmitters and all affecting one another. 

The cleanest way I found to handle it was to implement the echo server as a Karma middleware plugin, delete `run.js`, and use `karma start` in the package.json script.